Source/Tools/LinuxEmulation/LinuxSyscalls/GdbServer.cpp

```diff
@@ -604,8 +600,11 @@ GdbServer::HandledPacketType GdbServer::handleProgramOffsets() {
 GdbServer::HandledPacketType GdbServer::ThreadAction(char action, uint32_t tid) {
   switch (action) {
   case 'c': {
+    auto Threads = SyscallHandler->TM.GetThreads();
+    for (auto& Thread : *Threads) {
+      Thread->ThreadSleeping.NotifyOne();
+    }
```

Comment:
Is it safe to iterate over all threads like this? Can the threads list change while this code is running?

Reply:
Technically unsafe. Grabbed the thread creation mutex while doing this.